From 09cf18aa163fc20ed134d086b948b61559de4421 Mon Sep 17 00:00:00 2001
From: Dmitry Muhomor <muhomor.dmitry@gmail.com>
Date: Sun, 3 Dec 2023 17:36:48 +0200
Subject: [PATCH] make persist.adb.tls_server.enable system property
 non-persistent

persist.adb.tls_server.enable sysprop enables persistent network ADB, which severely weakens
verified boot.

Network ADB is disabled after each reboot by the system_server, but in a fragile way, see
frameworks/base/services/core/java/com/android/server/adb/AdbService.java

It's not clear whether this system_server behavior is intentional.
---
 init/property_service.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/init/property_service.cpp b/init/property_service.cpp
index f2606e3c5888..64a7f78da1e9 100644
--- a/init/property_service.cpp
+++ b/init/property_service.cpp
@@ -1455,6 +1455,9 @@ static void HandleInitSocket() {
             for (const auto& property_record : persistent_properties.properties()) {
                 auto const& prop_name = property_record.name();
                 auto const& prop_value = property_record.value();
+                if (prop_name == "persist.adb.tls_server.enable") {
+                    continue;
+                }
                 InitPropertySet(prop_name, prop_value);
             }
 
